core/src/main/java/org/bitcoinj/core/Coin.java

```diff
@@ -118,6 +118,36 @@ public static Coin valueOf(final int coins, final int cents) {
         return coin;
     }
 
+    /**
+     * Convert a decimal amount of bitcoin into satoshis.
+     *
+     * @param coins number of coins
+     * @return number of satoshis
+     */
+    public static long bitcoinToSatoshi(BigDecimal coins) {
+        return coins.movePointRight(SMALLEST_UNIT_EXPONENT).longValueExact();
+    }
+
+    /**
+     * Convert an amount in satoshis to an amount in bitcoin.
+     *
+     * @param satoshis number of satoshis
+     * @return number of bitcoins
+     */
+    public static BigDecimal satoshiToBitcoin(long satoshis) {
+        return new BigDecimal(satoshis).movePointLeft(SMALLEST_UNIT_EXPONENT);
+    }
+
+    /**
+     * Convert a decimal amount of bitcoin into satoshis.
+     *
+     * @param coins number of coins
+     * @return {@code Coin} object containing value in satoshis
+     */
+    public static Coin ofBitcoin(BigDecimal coins) {
```

Comment:
I think this should be named `valueOf` too, for consistency.

Reply:
I think it needs to have a different name because it takes a "different" argument.

    long satoshis = 50000000
    BigDecimal bitcoin = new BigDecimal(satoshis)
    Coin coin = Coin.valueOf(bitcoin)

What is the value of `coin` at this point?

Because both methods take `Number` types but using different units (orders of magnitude) it doesn't seem intuitive to use method overloading (i.e. the same name with different types) of the same (static) method name.

Ideally, I'd have:

    public static Coin of(long satoshis)
    public static Coin ofBitcoin(BigDecimal bitcoin)

In theory we could also have:

    public static Coin of(BigDecimal satoshis)
    public static Coin ofBitcoin(long bitcoin)

but I don't think those methods are really necessary.

Reply:
I see your point. I would make the denomination clearer however:

```
public static Coin ofSat(long satoshis)
public static Coin ofBtc(BigDecimal bitcoin)
```

Yes, "bitcoin" is also a denomination, but it's also a currency especially when written with an uppercase B (which we're forced to due to camel-casing).